packages/discord.js/src/structures/Guild.js

```diff
@@ -788,51 +787,59 @@ class Guild extends AnonymousGuild {
-  async edit(options) {
-    const _data = {};
-    if (options.name) _data.name = options.name;
-    if (typeof options.verificationLevel !== 'undefined') {
-      _data.verification_level = options.verificationLevel;
-    }
-    if (typeof options.afkChannel !== 'undefined') {
-      _data.afk_channel_id = this.client.channels.resolveId(options.afkChannel);
-    }
-    if (typeof options.systemChannel !== 'undefined') {
-      _data.system_channel_id = this.client.channels.resolveId(options.systemChannel);
-    }
-    if (options.afkTimeout) _data.afk_timeout = Number(options.afkTimeout);
-    if (typeof options.icon !== 'undefined') _data.icon = await DataResolver.resolveImage(options.icon);
-    if (options.owner) _data.owner_id = this.client.users.resolveId(options.owner);
-    if (typeof options.splash !== 'undefined') _data.splash = await DataResolver.resolveImage(options.splash);
-    if (typeof options.discoverySplash !== 'undefined') {
-      _data.discovery_splash = await DataResolver.resolveImage(options.discoverySplash);
-    }
-    if (typeof options.banner !== 'undefined') _data.banner = await DataResolver.resolveImage(options.banner);
-    if (typeof options.explicitContentFilter !== 'undefined') {
-      _data.explicit_content_filter = options.explicitContentFilter;
-    }
-    if (typeof options.defaultMessageNotifications !== 'undefined') {
-      _data.default_message_notifications = options.defaultMessageNotifications;
-    }
-    if (typeof options.systemChannelFlags !== 'undefined') {
-      _data.system_channel_flags = SystemChannelFlagsBitField.resolve(options.systemChannelFlags);
-    }
-    if (typeof options.rulesChannel !== 'undefined') {
-      _data.rules_channel_id = this.client.channels.resolveId(options.rulesChannel);
-    }
-    if (typeof options.publicUpdatesChannel !== 'undefined') {
-      _data.public_updates_channel_id = this.client.channels.resolveId(options.publicUpdatesChannel);
-    }
-    if (typeof options.features !== 'undefined') {
-      _data.features = options.features;
-    }
-    if (typeof options.description !== 'undefined') {
-      _data.description = options.description;
-    }
-    if (typeof options.preferredLocale !== 'undefined') _data.preferred_locale = options.preferredLocale;
-    if ('premiumProgressBarEnabled' in options) _data.premium_progress_bar_enabled = options.premiumProgressBarEnabled;
-    const newData = await this.client.rest.patch(Routes.guild(this.id), { body: _data, reason: options.reason });
-    return this.client.actions.GuildUpdate.handle(newData).updated;
+  async edit({
+    name,
+    verificationLevel,
+    defaultMessageNotifications,
+    explicitContentFilter,
+    afkChannel,
+    afkTimeout,
+    icon,
+    owner,
+    splash,
+    discoverySplash,
+    banner,
+    systemChannel,
+    systemChannelFlags,
+    rulesChannel,
+    publicUpdatesChannel,
+    preferredLocale,
+    features,
+    description,
+    premiumProgressBarEnabled,
+    reason,
```

Comment:
There are some fields that are as-is with Discord's, we can rest those an map only the camelCase values (and `reason`).

Those would be `name`, `features`, and `description`.

Reply:
Should be done with f9dcd8d5745d2a2cc9ec715b284dc7180799606d.